src/writer/writer-json.ts

```diff
@@ -7,14 +8,38 @@ export interface WriteJsonOptions {
   input: string;
   inputDir: string;
   outFile: string;
+  outDir?: string;
 }
 
 interface JsonOutput {
+  module?: string;
   total: number;
   components: ComponentDocApi[];
 }
 
-export default async function writeJson(components: ComponentDocs, options: WriteJsonOptions) {
+async function writeJsonComponents(components: ComponentDocs, options: WriteJsonOptions) {
+  debugger;
+  const output = Array.from(components, ([moduleName, component]) => ({
+    ...component,
+    filePath: normalizeSeparators(path.join(options.inputDir, path.normalize(component.filePath))),
+  })).sort((a, b) => {
+    if (a.moduleName < b.moduleName) return -1;
+    if (a.moduleName > b.moduleName) return 1;
+    return 0;
+  });
+
+  output.map((c) => {
+    const outFile = path.resolve(
+      path.join(dirname(options.outDir as string), `${c.moduleName.toLowerCase()}-api.json`)
```

Comment:
When testing this locally, the `dirname` wasn't working as expected, I actually don't think it's needed if the path is being resolved.

```suggestion
      path.join(options.outDir || "", `${c.moduleName.toLowerCase()}-api.json`));
```